src/main/java/de/tum/cit/aet/artemis/exercise/api/ExerciseApi.java

```diff
@@ -0,0 +1,84 @@
+package de.tum.cit.aet.artemis.exercise.api;
+
+import java.util.Set;
+
+import org.springframework.stereotype.Controller;
+
+import de.tum.cit.aet.artemis.assessment.domain.ExampleSubmission;
+import de.tum.cit.aet.artemis.assessment.domain.GradingCriterion;
+import de.tum.cit.aet.artemis.assessment.repository.ExampleSubmissionRepository;
+import de.tum.cit.aet.artemis.assessment.repository.GradingCriterionRepository;
+import de.tum.cit.aet.artemis.communication.domain.conversation.Channel;
+import de.tum.cit.aet.artemis.communication.repository.conversation.ChannelRepository;
+import de.tum.cit.aet.artemis.core.domain.User;
+import de.tum.cit.aet.artemis.core.service.AuthorizationCheckService;
+import de.tum.cit.aet.artemis.exercise.domain.Exercise;
+import de.tum.cit.aet.artemis.exercise.service.ExerciseService;
+
+// This is a very simple implementation, and should pbl be split up
+@Controller
+public class ExerciseApi {
+
+    // ToDo: Should be some kind of ChannelApi
+    private final ChannelRepository channelRepository;
+
+    // ToDo: Should be some kind of AssessmentApi
+    private final GradingCriterionRepository gradingCriterionRepository;
+
+    // ToDo: Should be some kind of AssessmentApi
+    private final ExampleSubmissionRepository exampleSubmissionRepository;
+
+    private final AuthorizationCheckService authCheckService;
+
+    private final ExerciseService exerciseService;
+
+    public ExerciseApi(AuthorizationCheckService authCheckService, ChannelRepository channelRepository, GradingCriterionRepository gradingCriterionRepository,
+            ExerciseService exerciseService, ExampleSubmissionRepository exampleSubmissionRepository) {
+        this.authCheckService = authCheckService;
+        this.channelRepository = channelRepository;
+        this.gradingCriterionRepository = gradingCriterionRepository;
+        this.exerciseService = exerciseService;
+        this.exampleSubmissionRepository = exampleSubmissionRepository;
+    }
+
+    public boolean isStudentAllowedToSee(Exercise exercise) {
+        return isStudentAllowedToSee(exercise, null);
+    }
+
+    public boolean isStudentAllowedToSee(Exercise exercise, User user) {
+        if (exercise.isExamExercise()) {
+            return false;
+        }
+
+        return authCheckService.isAllowedToSeeExercise(exercise, user);
+    }
+
+    public boolean isCorrectExerciseType(Exercise exercise, Class<? extends Exercise> expectedExerciseClass) {
+        return expectedExerciseClass.isInstance(exercise);
+    }
+
+    public void setChannelName(Exercise exercise) {
```

Comment:
We might want to improve the naming here. For me, it's something similar to `hydrateWithChannelName` or `loadChannelName` to express that we are loading something from the DB.